athena-slack-member-analytics/src/main/java/com/amazonaws/connectors/athena/slack/util/SlackHttpUtility.java

```diff
@@ -0,0 +1,188 @@
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.zip.GZIPInputStream;
+
+public class SlackHttpUtility {
+
+    private static final Logger logger = LoggerFactory.getLogger(SlackHttpUtility.class);
+
+    /**
+     * Makes an HTTP request using GET method to the specified URL.
+     *
+     * @param requestURL
+     *            the URL of the remote server
+     * @param headers HashMap<String,String> of request Headers
+     * @return An CloseableHttpResponse object
+     * @throws IOException
+     *             thrown if any I/O error occurred
+     */
+    public static CloseableHttpResponse doGetRequest(URIBuilder requestURI, HashMap<String, String> headers)
+            throws IOException, Exception {
+
+        logger.info("doGetRequest: enter - {}", requestURI.toString());
+        HttpGet httpGet = new HttpGet(requestURI.build());
+        for (String i : headers.keySet()) {
+            httpGet.setHeader(i,headers.get(i));
+        }
+
+        CloseableHttpClient client = HttpClients.createDefault();
+        CloseableHttpResponse response = client.execute(httpGet);
+        if (!isRequestOk(response)){
+            client.close();
+            throw new IOException("doGetRequest: response " + response.getStatusLine().getReasonPhrase());
+        }
+
+        logger.info("doGetRequest: exit");
+        return response;
+
+    }
+
+    /**
+     * Makes an HTTP request using POST method to the specified URL.
+     *
+     * @param requestURL the URL of the remote server
+     * @param headers HashMap<String,String> of request Headers
+     * @param params HashMap<String, String> of post parameters
+     * @return An CloseableHttpResponse object
+     * @throws IOException
+     *             thrown if any I/O error occurred
+     */
+    public static CloseableHttpResponse doPostRequest(String requestURL,
+                                                      HashMap<String, String> params,
+                                                      HashMap<String, String> headers)
+            throws Exception {
+
+        logger.info("doPostRequest: enter - {}", requestURL);
+        HttpPost httpPost = new HttpPost(requestURL);
+
+        if (params != null && params.size() > 0) {
+            List<NameValuePair> nameValuePairs = new ArrayList<>(params.size());
+            for (String i : params.keySet()){
+                logger.debug("doPostRequest - params {}:{}", i, params.get(i));
+                nameValuePairs.add(new BasicNameValuePair(i,params.get(i)));
+            }
+            httpPost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
+        }
+
+        if(headers != null && headers.size()>0){
+            for (String i : headers.keySet()) {
+                logger.debug("doPostRequest - headers {}:{}", i, headers.get(i));
+                httpPost.setHeader(i,headers.get(i));
+            }
+        }
+        CloseableHttpClient client = HttpClients.createDefault();
```

Comment:
same comment about client not always being closed

Reply:
closing response object instead of client